packages/labs/test/tagInputTests.tsx

```diff
@@ -376,6 +376,50 @@ describe("<TagInput>", () => {
         });
     });
 
+    describe("onInputChange", () => {
+        it("is not invoked on enter when input is empty", () => {
+            const onInputChange = sinon.stub();
+            const wrapper = shallow(<TagInput onInputChange={onInputChange} values={VALUES} />);
+            pressEnterInInput(wrapper, "");
+            assert.isTrue(onInputChange.notCalled);
+        });
+
+        it("is invoked when input text changes", () => {
+            const changeSpy: any = sinon.spy();
+            const wrapper = shallow(<TagInput onInputChange={changeSpy} values={VALUES} />);
+            wrapper.find("input").simulate("change", { currentTarget: { value: "hello" } });
+            assert.isTrue(changeSpy.calledOnce, "onChange called");
+            assert.equal("hello", changeSpy.args[0][0].currentTarget.value);
+        });
+    });
+
+    describe("inputValue", () => {
+        const NEW_VALUE = "new item";
+        it("passes initial inputValue to input element", () => {
+            const input = shallow(<TagInput values={VALUES} inputValue={NEW_VALUE} />).find("input");
+            assert.isTrue(input.prop("value") === NEW_VALUE);
+        });
+
+        it("accepts successive input value changes", () => {
+            const wrapper = mount(<TagInput values={VALUES} />);
+            wrapper.setProps({ inputValue: "a" });
+            expect(wrapper.state().inputValue).to.equal("a");
+            wrapper.setProps({ inputValue: "b" });
+            expect(wrapper.state().inputValue).to.equal("b");
+            wrapper.setProps({ inputValue: "c" });
+            expect(wrapper.state().inputValue).to.equal("c");
+        });
+
+        it("Updating inputValue updates input element", () => {
+            const wrapper = mount(<TagInput values={VALUES} />);
+            const input = wrapper.find("input");
```

Comment:
don't cache the input before you set props, just do the `find` in the `assert` on line 419.

Reply:
Done